api/src/main/java/org/apache/iceberg/ManifestFile.java

```diff
@@ -167,4 +174,11 @@ default boolean hasDeletedFiles() {
      */
     PartitionFieldSummary copy();
   }
+
+  // TODO: consider other approaches such as adding builders directly to GenericManifestFile
+  interface CopyBuilder {
```

Comment:
The goal is to provide a generic way to enrich manifests with additional metadata without creating a new object every time we modify a field (snapshot id, sequence number). I've considered multiple ways to implement this. My original idea was to add a copy builder to `GenericManifestFile`. 

```
GenericManifestFile.copy(manifest)
    .withSnapshotId(snapshotId)
    .build();
```

Then I thought it would make sense to rely on the abstract interface instead. After I implemented this, the original idea seems not that bad as it allows us to keep `ManifestFile` untouched. I consider switching back.

Reply:
Yeah, I'd probably prefer leaving this untouched and keeping the builder in core rather than api.